File: dotnet/src/Model/Neutral/Buildings/Building.cs
using System;

namespace PlanetCS1 {
    public abstract class Building : RealTile {
        public Building() {
            this.occ = true;
            this.drive = false;
            this.traverse = false;
            this.minSal = 0;
            this.gasSal = 0;
            this.nrgSal = 0;
        }
    }
}

File: jvm2/src/Model/Neutral/Buildings/Building.kt
package Model.Neutral.Buildings

import Model.Abstract.RealTile

abstract class Building() : RealTile() {
    override var occ = true
    override var drive = false
    override var traverse = false
    override var minSal = 0
    override var gasSal = 0
    override var nrgSal = 0
}